File: 2015/Day08/Day8.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <algorithm>
#include <array>

using namespace std;

void removeEscSeq(string& s);

int main(){

	ifstream inputFile ("input.txt");
	string line;
	int nCode = 0;
	int nMem = 0;

	while (getline(inputFile, line)){

		nCode += line.length();
		removeEscSeq(line);
		nMem += line.length();
	}

	cout << "Answer: " << nCode - nMem << endl;
}


void removeEscSeq(string& s){

	array<string,16> escSeq {"\\a", "\\b", "\\e", "\\f", "\\n", "\\r", "\\t", "\\v",
                             "\\\\", "\\\'", "\\\"", "\\?", "\\nnn", "\\xhh…", 
                             "\\uhhhh", "\\Uhhhhhhhh"};

	for (auto it = escSeq.begin(); it != escSeq.end(); ++it){

		string::size_type i = s.find(*it);

    	while (i != string::npos) {

    		s.erase(i, (*it).length());
    		i = s.find(*it, i);
    	}
	}
}

File: 2015/Day08/Day8.py
def main():

	nCode = 0
	nMem = 0
	nNewCode = 0

	with open("input.txt", "r") as input_file:
		lines = input_file.read().splitlines()
		for line in lines:
			nCode += len(line)
			nMem += len(eval(line))
			nNewCode += len(line) + 2 + line.count('\\') + line.count('\'') + line.count('\"')

	print("Answer to part one =", nCode - nMem)
	print("Answer to part two =", nNewCode - nCode)

main()